avgratecounter.go

```diff
@@ -2,39 +2,46 @@ package ratecounter
 
 import (
 	"strconv"
-	"sync/atomic"
 	"time"
 )
 
 // An AvgRateCounter is a thread-safe counter which returns
 // the ratio between the number of calls 'Incr' and the counter value in the last interval
 type AvgRateCounter struct {
-	hits     int64
-	counter  Counter
+	hits     *RateCounter
+	counter  *RateCounter
 	interval time.Duration
 }
 
 // NewRateCounter Constructs a new AvgRateCounter, for the interval provided
 func NewAvgRateCounter(intrvl time.Duration) *AvgRateCounter {
 	return &AvgRateCounter{
+		hits:     NewRateCounter(intrvl),
+		counter:  NewRateCounter(intrvl),
 		interval: intrvl,
 	}
 }
 
+func (r *AvgRateCounter) WithResolution(resolution int) *AvgRateCounter {
```

Comment:
Again, let's do:
`func NewAvgRateCounter(intrvl time.Duration, resolution int) *AvgRateCounter`
and just do it all at once.